### ISSUE TITLE
Read-only does not work for gpt_ext4

### ISSUE THREAD

**Issue by drewmoseley** [systemd/mkosi#543]
Specifying readonly does not take effect with at least the ext4 partitions.

Using the following config:
```
$ \cat mkosi.default 
[Distribution]
Distribution=ubuntu
Release=focal

[Output]
Format=raw_gpt
Bootable=yes
Incremental=yes
ReadOnly=True
```

Then after booting with qemu:
```
root@localhost:~# mount | grep sda
/dev/sda2 on / type ext4 (rw,relatime)
root@localhost:~# touch /foo.txt
root@localhost:~# ls -ld /foo.txt
-rw-r--r-- 1 root root 0 Dec  1 18:10 /foo.txt
```

An initial attempt at a [fix is here](https://github.com/systemd/mkosi/pull/541).

**Comment by behrmann** [comment]
Ah, yes. I see your problem, that is clearly a bug, since the CLI flag advertises this functionality.

Looking further at the code, my initial comment about this only being for verity and squashfs was wrong. A couple of question on your example:
- Which version / commit of mkosi was used to generate this?
- Does an `/etc/fstab` exist? If so, what are its contents? 
- What is the content of `/proc/cmdline`?

If `/etc/fstab` exists and `/` is mounted with `rw` or `defaults`, this is the problem, I think, because the explicit fstab would trump the flag that would only be picked up by the gpt-auto-generator. If no fstab exists, could you have a look with [gdisk](https://wiki.archlinux.org/index.php/GPT_fdisk#Prevent_GPT_partition_automounting), whether bit 60 is set in the attributes?

**Comment by drewmoseley** [comment]
I am on tag v7 (5c19c9aee7aa19bb2072c4a57389e00ab3d29618)

The fstab is simply `# UNCONFIGURED FSTAB FOR BASE SYSTEM`

/proc/cmdline is `rhgb quiet selinux=0 audit=0 rw`

**Comment by drewmoseley** [comment]
gdisk shows `Attribute flags: 1000000000000000` on the root partition.

**Comment by drewmoseley** [comment]
I modified my sources to remove "rw" from the default kernel_command_line setting and that seemed to be enough for this to work. If that's a reasonable solution I'll update the PR.

**Comment by behrmann** [comment]
> I modified my sources to remove "rw" from the default kernel_command_line setting and that seemed to be enough for this to work. If that's a reasonable solution I'll update the PR.

I think this is a sensible idea. Very good.

**Comment by DaanDeMeyer** [comment]
Fixed by #541
